samples/snippets/src/main/java/pubsublite/CreateSubscriptionExample.java

```diff
@@ -41,27 +41,55 @@ public static void main(String... args) throws Exception {
     String topicId = "your-topic-id";
     String subscriptionId = "your-subscription-id";
     long projectNumber = Long.parseLong("123456789");
+    boolean regional = false;
 
-    createSubscriptionExample(cloudRegion, zoneId, projectNumber, topicId, subscriptionId);
+    createSubscriptionExample(
+        cloudRegion, zoneId, projectNumber, topicId, subscriptionId, regional);
   }
 
   public static void createSubscriptionExample(
-      String cloudRegion, char zoneId, long projectNumber, String topicId, String subscriptionId)
+      String cloudRegion,
+      char zoneId,
+      long projectNumber,
+      String topicId,
+      String subscriptionId,
+      boolean regional)
       throws Exception {
 
-    TopicPath topicPath =
-        TopicPath.newBuilder()
-            .setProject(ProjectNumber.of(projectNumber))
-            .setLocation(CloudZone.of(CloudRegion.of(cloudRegion), zoneId))
-            .setName(TopicName.of(topicId))
-            .build();
+    TopicPath topicPath = null;
+    SubscriptionPath subscriptionPath = null;
 
-    SubscriptionPath subscriptionPath =
-        SubscriptionPath.newBuilder()
-            .setLocation(CloudZone.of(CloudRegion.of(cloudRegion), zoneId))
-            .setProject(ProjectNumber.of(projectNumber))
-            .setName(SubscriptionName.of(subscriptionId))
-            .build();
+    if (regional) {
```

Comment:
You can reduce duplication here by instead assigning to a CloudRegionOrZone variable.
CloudRegionOrZone location;
if (regional) {
   location = CloudRegionOrZone.of(CloudRegion.of(cloudRegion));
} else {
   location = CloudRegionOrZone.of(CloudZone.of(CloudRegion.of(cloudRegion), zoneId));
}

Reply:
Or you could have a local TopicPath.Builder and SubscriptionPath.Builder object and do:

```
TopicPath.Builder topicBuilder = ...
SubscriptionPath.Builder subscriptionBuilder = ...
if (regional) {
  topicBuilder.setLocation(CloudRegion.of(cloudRegion));
  subscriptionBuilder.setLocation(CloudRegion.of(cloudRegion));
} else {
  ...
}
```

Reply:
Ack. I considered it, but did not show this convenient class.
We also do not show `ProjectNumberOrId`.

Reply:
Reducing duplication with `CloudRegionOrZone`.